spec/requests/reviewable_notes_controller_spec.rb

```diff
@@ -0,0 +1,323 @@
+      before { sign_in(admin) }
+
+      it "returns 404" do
+        delete "/reviewables/999999/notes/#{note.id}.json"
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context "when note belongs to different reviewable" do
+      let(:other_reviewable) { Fabricate(:reviewable_flagged_post) }
+      let!(:other_note) { Fabricate(:reviewable_note, reviewable: other_reviewable, user: admin) }
+
+      before { sign_in(admin) }
+
+      it "returns 404" do
+        delete "/reviewables/#{reviewable.id}/notes/#{other_note.id}.json"
+        expect(response.status).to eq(404)
+      end
+    end
+  end
+
+  describe "authorization" do
+    context "when testing ensure_staff callback" do
+      it "blocks non-staff users from accessing create" do
+        sign_in(user)
+
+        post "/reviewables/#{reviewable.id}/notes.json",
+             params: {
+               reviewable_note: {
+                 content: "Test",
+               },
+             }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "blocks non-staff users from accessing destroy" do
+        note = Fabricate(:reviewable_note, reviewable: reviewable, user: admin)
+        sign_in(user)
+
+        delete "/reviewables/#{reviewable.id}/notes/#{note.id}.json"
+
+        expect(response.status).to eq(403)
+      end
+    end
+
+    context "when testing requires_login callback" do
+      it "blocks anonymous users from accessing create" do
+        post "/reviewables/#{reviewable.id}/notes.json",
+             params: {
+               reviewable_note: {
+                 content: "Test",
+               },
+             }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "blocks anonymous users from accessing destroy" do
+        note = Fabricate(:reviewable_note, reviewable: reviewable, user: admin)
+
+        delete "/reviewables/#{reviewable.id}/notes/#{note.id}.json"
+
+        expect(response.status).to eq(403)
+      end
+    end
+  end
+
+  describe "parameter handling" do
+    before { sign_in(admin) }
+
+    it "requires reviewable_note parameter for create" do
+      post "/reviewables/#{reviewable.id}/notes.json", params: { content: "Missing wrapper param" }
+
+      expect(response.status).to eq(400)
+    end
+
+    it "only permits content parameter" do
+      post "/reviewables/#{reviewable.id}/notes.json",
+           params: {
+             reviewable_note: {
+               content: "Valid content",
+               user_id: user.id, # Should be ignored
+               reviewable_id: 999, # Should be ignored
+               created_at: 1.day.ago, # Should be ignored
+             },
+           }
+
+      expect(response.status).to eq(200)
+
+      note = ReviewableNote.last
+      expect(note.content).to eq("Valid content")
+      expect(note.user).to eq(admin) # Not the user_id from params
+      expect(note.reviewable).to eq(reviewable) # Not the reviewable_id from params
+    end
+  end
+end
```

Comment:
```suggestion
```

I don't feel like the tests are necessary since the tests for each controller actions already covers what these tests cover.